File: src/linkedlist/lc_2_add_two_numbers/add_two_numbers.java
/** Definition for singly-linked list.*/
 public class ListNode {
     int val;
     ListNode next;
     ListNode() {}
     ListNode(int val) { this.val = val; }
     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}
 
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode pt1, pt2;
        ListNode dummy = new ListNode(0);
        ListNode curr = dummy;
        pt1 = l1;
        pt2 = l2;
        int rem = 0;
        while (pt1 != null && pt2 != null) {
            int val = pt1.val + pt2.val + rem;
            if (val > 9) {
                val = val % 10;
                rem = 1;
            } else {
                rem = 0;
            }
            
            ListNode node = new ListNode(val);
            curr.next = node;
            curr = curr.next;
            pt1 = pt1.next;
            pt2 = pt2.next;
        }
        while (pt1 != null) {
            int val = pt1.val + rem;
            if (val > 9) {
                val = val % 10;
                rem = 1;
            } else {
                rem = 0;
            }
            ListNode node = new ListNode(val);
            curr.next = node;
            curr = curr.next;
            pt1 = pt1.next;
        }
        while (pt2 != null) {
            int val = pt2.val + rem;
            if (val > 9) {
                val = val % 10;
                rem = 1;
            } else {
                rem = 0;
            }
            ListNode node = new ListNode(val);
            curr.next = node;
            curr = curr.next;
            pt2 = pt2.next;
        }
        if (rem == 1) {
            curr.next = new ListNode(1, null);
        }
        return dummy.next;
    }
}

File: src/linkedlist/lc_2_add_two_numbers/add_two_numbers.py
from listNode import ListNode

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        dummy = ListNode(0, None)
        temp = dummy
        t1, t2 = l1, l2
        rem = 0

        while t1 and t2:
            v1 = t1.val
            v2 = t2.val
            v = v1 + v2
            if rem > 0:
                v += rem
            if v // 10 > 0:
                rem = 1
                v = v % 10
            else:
                rem = 0
            
            newNode = ListNode(v, None)
            temp.next = newNode
            temp = temp.next
            t1 = t1.next
            t2 = t2.next
        while t1:
            v = t1.val
            if rem > 0:
                v += rem
            if v // 10 > 0:
                rem = 1
                v = v % 10
            else:
                rem = 0
            newNode = ListNode(v, None)
            temp.next = newNode
            temp = temp.next
            
            t1 = t1.next
        while t2:
            v = t2.val
            if rem > 0:
                v += rem
            if v // 10 > 0:
                rem = 1
                v = v % 10
            else:
                rem = 0
            newNode = ListNode(v, None)
            temp.next = newNode
            temp = temp.next
            
            t2 = t2.next
        if rem > 0:
            newNode = ListNode(rem, None)
            temp.next = newNode
        return dummy.next
        
           